tiledb/cf/core.py

```diff
@@ -353,28 +328,54 @@ def create_virtual(
                 decrypt arrays.
             ctx: If not ``None``, TileDB context wrapper for a TileDB storage manager.
         """
-        cls.create(uri, array_schemas, group_metadata, key, ctx, is_virtual=True)  # pragma: no cover
+        cls.create(uri, group_schema, key, ctx, is_virtual=True)  # pragma: no cover
 
     def __init__(
         self,
         uri: str,
+        mode: str = "r",
         key: Optional[Union[Dict[str, str], str]] = None,
         timestamp: Optional[int] = None,
+        array: Optional[str] = None,
         attr: Optional[str] = None,
         ctx: Optional[tiledb.Ctx] = None,
     ):
         """Constructs a new :class:`Group`."""
-        self.uri = uri
-        self.key = key
-        self.timestamp = timestamp
-        self.ctx = ctx
-        self.open_arrays = dict()
-
-        self._attr = attr
+        self._group_schema = GroupSchema.load(uri, ctx, key)
+        self._array_uris = {
```

Comment:
can we get the uri from the name which is provided in open_array?

Reply:
I did it this way to make accessing the URI the same for both the `Group` and the `VirtualGroup` classes. Otherwise, you would need to change the implementation of `open_array` in the `VirtualGroup` class.